sparse/impl/KokkosSparse_par_ilut_symbolic_impl.hpp

```diff
@@ -44,6 +44,8 @@ void ilut_symbolic(IlutHandle& thandle, const ARowMapType& A_row_map_d,
   using size_type       = typename IlutHandle::size_type;
   using Ilut            = IlutWrap<IlutHandle>;
 
+  thandle.set_nrows(A_row_map_d.extent(0) - 1);
```

Comment:
We also have to allow 0-length rowmaps for 0-row matrices (this case actually does come up in Ifpack2, since Tpetra will generally not allocate a 1-element rowmap for a 0-row matrix)

Reply:
I also saw below that par_ilut_symbolic requires that A, L, and U rowmaps all have the same extent. I think it would be preferable to let those differ, and only require that the numbers of rows match. For example, in your unit test you do:

```
// Allocate L and U CRS views as outputs
RowMapType L_row_map("L_row_map", numRows + 1);
RowMapType U_row_map("U_row_map", numRows + 1);
```
which is I think what most people would do, and it's correct, but it would trigger this check when ``A_rowmap.extent(0) == 0``

Reply:
@brian-kelley , I'm a bit confused how would I check number of rows match without checking extents? Also, are you saying A.extent(0)=0, L.extent(0)=1, U.extent(0)=1 is an expected use case?

Reply:
@jgfouca You still get it from the rowmap extent, just with this special case covered:
```
numRows = rowmap.extent(0) ? (rowmap.extent(0) - 1) : size_t(0);
```
And yes, that use case should work. It wouldn't be a bad idea to even test the zero-rows case in unit tests like gauss-seidel does